app.js

```diff
@@ -0,0 +1,23 @@
+const express = require("express");
+const {
+    getCategories,
+} = require("./controllers/controllers");
+
+const app = express();
+
+app.get('/api/categories', getCategories);
+
+app.all("*", (request, response)=>{
+    response.status(404).send({status: 404, msg: "Not Found"})
+})
+
+app.use((error, request, response, next)=>{
+    if(error.hasOwnProperty("status") && error.hasOwnProperty("msg"))
+    response.status(error.status).send({status: error.status, msg: error.msg})
+})
+
+app.use((error, request, response, next)=>{
+    response.status(500).send({ msg: "this link does not exist"})
```

Comment:
I would say that a 500 status is not that something doesnt exist, it is an Internal Server Error, this essentially means we have no idea what caused this and as such, have not accounted for it with a different status code. I would leave this as Internal Server Error for the message. You may also benefit from having a console.log(error) in this error handler as it will catch all the errors you havent accounted for yet and log them for you when they happen so you know how to deal with it.